src/actions/Actions.ts

```diff
@@ -177,94 +189,102 @@ export const AsyncActions = {
-                        .slice(0, 20)
-                        .map(p => ({
-                            ...p,
-                            images: p.images.map(image => ({
-                                ...image,
-                                localPath: undefined,
-                            })),
-                        }))
-                        ;
+                dispatch(Actions.updateOwnFeed({
+                    ...localFeed,
+                    isSyncing: true,
+                }));
 
-                    const uploadedPosts = await uploadPosts(posts, resizeImageIfNeeded, modelHelper);
-                    const postFeed = {
-                        ...feed,
-                        posts: uploadedPosts,
-                        authorImage: {
-                            ...feed.authorImage,
-                            localPath: '',
-                        },
-                    };
-                    Debug.log('sharePost: after uploadPosts');
+                const identity = getState().author.identity!;
+                const signFeedDigest = (digest: number[]) => Swarm.signDigest(digest, identity);
+                const swarmStorageSyncer = getSwarmStorageSyncer(signFeedDigest, localFeed);
 
-                    await updatePostFeed(swarmFeedApi, postFeed);
-                    Debug.log('sharePost: after uploadPostFeed');
+                const localPostCommandLog = localFeed.postCommandLog;
+                const storageSyncUpdate = await swarmStorageSyncer.sync(localPostCommandLog, localFeed);
+                Debug.log('uploadPostFromQueue', 'storageSyncUpdate', storageSyncUpdate);
 
-                    dispatch(Actions.updatePostLink(post, feed.url));
-                    dispatch(Actions.updatePostIsUploading(post, undefined));
-
-                    const mergedImages = mergeImages(post.images, uploadedPost.images);
-                    dispatch(Actions.updatePostImages(post, mergedImages));
-                } else {
-                    Debug.log('sharePost: create feed');
+                storageSyncUpdate.updatedPosts.map(updatedPost => {
+                    dispatch(Actions.updatePostLink(updatedPost, localFeed.url));
+                    dispatch(Actions.updatePostIsUploading(updatedPost, undefined));
+                    const localPosts = getState().localPosts;
+                    const originalPost = localPosts.find(p => p._id === updatedPost.author);
+                    if (originalPost != null) {
+                        const mergedImages = mergeImages(originalPost.images, updatedPost.images);
+                        dispatch(Actions.updatePostImages(updatedPost, mergedImages));
+                    }
+                    return updatedPost;
+                });
 
-                    const author = getState().author;
-                    dispatch(Actions.updatePostIsUploading(post, true));
+                // Re-check if there were an update to the command log during syncing
```

Comment:
will this still be an issue without gateway? (running swarm on mobile)

Reply:
Yes, it's still an issue, just the probability is smaller. Because we wait after every feed update  one second, and there are minimum 2 feed updates when syncing, it still allows enough time to accidentally run into this and the result is a post that is stuck in the spinning state.